src/couch_replicator/src/couch_replicator_utils.erl

```diff
@@ -276,6 +278,98 @@ seq_encode(Seq) ->
     % object. We are being maximally compatible here.
     ?JSON_ENCODE(Seq).
 
+%% Log uses of http protocol and uses of https protocol where verify_peer
+%% would fail.
+log_security_warnings(#rep{} = Rep) ->
+    case config:get_boolean("replicator", "log_security_warnings", false) of
+        false ->
+            ok;
+        true ->
+            ok = check_security(Rep, source),
+            ok = check_security(Rep, target)
+    end.
+
+check_security(#rep{} = Rep, Type) ->
+    Url =
+        case Type of
+            source -> Rep#rep.source#httpdb.url;
+            target -> Rep#rep.target#httpdb.url
+        end,
+    #url{protocol = Protocol} = ibrowse_lib:parse_url(Url),
+    case Protocol of
+        http ->
+            couch_log:warning("**security warning** replication ~s has insecure ~s at ~s", [
+                rep_principal(Rep), Type, Url
+            ]),
+            ok;
+        https ->
+            VerifyEnabled = config:get_boolean("replicator", "verify_ssl_certificates", false),
+            CACertFile = config:get("replicator", "ssl_trusted_certificates_file"),
```

Comment:
couch_replicator_parse parses, validates these options already, it might be possible read those options instead of re-parsing everything from scratch.

Reply:
I think you mean I should look at `ssl_options` from #rep.options which is constructed as 

```
VerifyCerts = cfg_boolean("verify_ssl_certificates", false),
ssl_verify_options(VerifyCerts)
```

```
-spec ssl_verify_options(true | false) -> [_].
ssl_verify_options(true) -> Used 1 times
    CAFile = cfg("ssl_trusted_certificates_file"),
    [{verify, verify_peer}, {cacertfile, CAFile}];
ssl_verify_options(false) ->
    [{verify, verify_none}].
```

But I need to detect when cacertfile _is_ set and verify_peer is _not_ set. Which is not possible from this.

Reply:
doesn't seem a worthwhile saving to me.

Reply:
fair enough